static/app/components/modals/widgetViewerModal.tsx

```diff
@@ -0,0 +1,168 @@
+import * as React from 'react';
+import {css} from '@emotion/react';
+import styled from '@emotion/styled';
+import cloneDeep from 'lodash/cloneDeep';
+
+import {ModalRenderProps} from 'sentry/actionCreators/modal';
+import {Client} from 'sentry/api';
+import Button from 'sentry/components/button';
+import ButtonBar from 'sentry/components/buttonBar';
+import {t} from 'sentry/locale';
+import {Organization, PageFilters} from 'sentry/types';
+import withApi from 'sentry/utils/withApi';
+import withPageFilters from 'sentry/utils/withPageFilters';
+import {DisplayType, Widget, WidgetType} from 'sentry/views/dashboardsV2/types';
+import {getFieldsFromEquations} from 'sentry/views/dashboardsV2/utils';
+import WidgetCardChartContainer from 'sentry/views/dashboardsV2/widgetCard/widgetCardChartContainer';
+
+export type WidgetViewerModalOptions = {
+  organization: Organization;
+  widget: Widget;
+};
+
+type Props = ModalRenderProps &
+  WidgetViewerModalOptions & {
+    api: Client;
+    organization: Organization;
+    selection: PageFilters;
+  };
+
+type State = {
+  loading: boolean;
+};
+
+const TABLE_ITEM_LIMIT = 30;
+const FULL_TABLE_HEIGHT = 600;
+const HALF_TABLE_HEIGHT = 300;
+
+class WidgetViewerModal extends React.Component<Props, State> {
```

Comment:
should we convert this to a regular function  component?

Reply:
Updated to a functional component!